src/cfe.c

```diff
@@ -601,6 +605,9 @@ void et_from_rainfall(double *timestep_rainfall_input_m, struct evapotranspirati
 //##############################################################
 void et_from_retention_depth(struct nash_cascade_parameters *nash_surface_params, struct evapotranspiration_structure *et_struct)
 {
+  // NJF try not to use nash_storage if it hasn't been initialized...
+  if(nash_surface_params->nash_storage != NULL) return;
+

```

Comment:
@hellkite500 given the condition on line 88, this (line 608) will always be true and we probably don't need it.

Reply:
This is *assuming* that `et_from_retention_depth` is never called from any other code, and/or that it is checked appropriately if it is.  This is more defensive than required, but since this function is a public function of the `cfe` library, I would strongly recommend this protection and/or additional documentation with this function to be clear about the result of using it without a properly allocated pointer in the structure.

Reply:
I see the logic here, however, my guess is you meant `if(nash_surface_params->nash_storage == NULL) return;` and not `if(nash_surface_params->nash_storage != NULL) return;`; see `==` vs `!=`. With `!=` it breaks the functionality of the extracting ET from the retention depth. you see what mean? happy to chat more if I am not clear

Reply:
Ah, yeah, I see what you are pointing out there, that is indeed the wrong logic! I'll push that fix real quick.

Reply:
fixed in [9270e6e](https://github.com/NOAA-OWP/cfe/pull/143/commits/9270e6e11910b74449efae734a7d19d4a9dd288a)